src/payment/payment-request-transformer.ts

```diff
@@ -20,13 +20,13 @@ export default class PaymentRequestTransformer {
     private _cardExpiryFormatter = new CardExpiryFormatter();
     private _cardNumberFormatter = new CardNumberFormatter();
 
-    transform(payment: Payment, checkoutState: InternalCheckoutSelectors): PaymentRequestBody {
+    transform(payment: Payment, checkoutState: InternalCheckoutSelectors, address?: Address | null): PaymentRequestBody {
         const billingAddress = checkoutState.billingAddress.getBillingAddress();
         const checkout = checkoutState.checkout.getCheckout();
         const customer = checkoutState.customer.getCustomer();
         const order = checkoutState.order.getOrder();
         const paymentMethod = checkoutState.paymentMethods.getPaymentMethod(payment.methodId, payment.gatewayId);
-        const shippingAddress = checkoutState.shippingAddress.getShippingAddress();
+        const shippingAddress = address === undefined ? checkoutState.shippingAddress.getShippingAddress() : (address ?? undefined);
```

Comment:
Do we need to do this, since `checkoutState.shippingAddress.getShippingAddress()` will return undefined if there is no address?

Reply:
@animesh1987 What I'm doing in this PR is definitely not the best solution of all, but I want you to understand what we need to see if you can suggest something more robust.

If BOPIS is enabled, each provider will behave differently. Some of them require not to send an address at all, some others require to send the pickup address. So, I need a way to dynamically set this `shippingAddress` const in order to achieve what we want. And also I need a way to reuse code because the same logic is required for googlepay strategy and applepay js 😬

In this case, for StripeV3, we need to send nothing for `shippingAddress`. So, I made the address param optional to avoid breaking anything and if `address === undefined` it means that it should behave like before and use `checkoutState.shippingAddress.getShippingAddress()`. For this case, I'm sending a `null` to differentiate from `undefined`. If I send a `null`, `shippingAddress` will be explicitly set to `undefined` and no shipping address will be send to Stripe.

I was thinking of injecting some kind of strategy into this `PaymentRequestTransformer` that contains the required logic for each provider...

Please, let me know if something is unclear.

Reply:
I think we can rethink the implementation in this case, since you can get paymentMethod and it's details from the payment passed. Also you have access to state. 

So I feel it maybe better to have a function here `_mapShippingAddress` or similar. And we can move the logic there. This avoids us to have the code in strategy, and then the need to pass an optional parameter down the method chain. Feel free to ping me on slack and we can work something out together.